guide/lifecycle-of-a-gradle-build.md

```diff
@@ -0,0 +1,80 @@
+<!-- PreviousNext:START -->
+<table><tr>
+  <td>Previous: <a href="managed-types-and-properties.md">Managed Types and Properties</a></td>
+  <td align="center"><a href="../README.md#table-of-contents">Table of Contents</a></td>
+  <td align="right">Next: <a href="avoiding-unnecessary-configuration.md">Avoiding Unnecessary Configuration</a></td>
+</tr></table>
+<!-- PreviousNext:END -->
+
+# Lifecycle of a Gradle Build
+
+> Further reading: [Gradle Build Lifecycle](https://docs.gradle.org/current/userguide/build_lifecycle.html)
+
+Gradle build have several stages, each of which are going to investigate in depth. As an overview, they are:
+
+1. Starting the build
+2. Settings evaluation
+3. Configuration phase
+4. Task dependency graph calculation
+5. Task execution
+
+## Starting the build
+
+When you _invoke_ gradle, you call the `./gradlew` script and provide a set of "requested tasks":
+
+```
+./gradlew taskA taskB
+```
+
+This `./gradlew` script runs the "Gradle Wrapper" - a new process that runs code in the jar checked into your repo at `gradle/wrapper/gradle-wrapper.jar`. The Gradle Wrapper will ensure the Gradle distribution listed in `gradle/wrapper/gradle-wrapper.properties` is installed - if it isn't, it will download and install it. [Further reading about the Gradle Wrapper](https://docs.gradle.org/current/userguide/gradle_wrapper.html).
+
+The Gradle Wrapper will now check to see if a "Gradle Daemon" already exists. Daemons are long-lived instances of Gradle which speed up build times by keeping the JVM running between builds. If a Daemon already exists, it will be reused. If not, a new one will be started. [Further reading about the Gradle Daemon](https://docs.gradle.org/current/userguide/gradle_daemon.html).
```

Comment:
is it worth talkign about daemon compatibility? i.e. you have daemons available but different jvm or some other weird thing is stopping it from being reused